validator.go

```diff
@@ -0,0 +1,67 @@
+package record
+
+import (
+	"errors"
+
+	logging "github.com/ipfs/go-log"
+)
+
+var log = logging.Logger("routing/record")
+
+// ErrBadRecord is returned any time a dht record is found to be
+// incorrectly formatted or signed.
+var ErrBadRecord = errors.New("bad dht record")
+
+// ErrInvalidRecordType is returned if a DHTRecord keys prefix
+// is not found in the Validator map of the DHT.
+var ErrInvalidRecordType = errors.New("invalid record keytype")
+
+// Validator is an interface that should be implemented by record validators.
+type Validator interface {
+	// Validate validates the given record, returning an error if it's
+	// invalid (e.g., expired, signed by the wrong key, etc.).
+	Validate(key string, value []byte) error
+
```

Comment:
Is it worth making the `Select` function look more like the `less` parameter to `Sort`'s [`Slice` function](https://golang.org/pkg/sort/#Slice)?
Otherwise every `Select` implementation will need to implement similar functionality to iterate over the values, return an error if the list is empty, or if there is no value found etc

Reply:
IMO, that can be implemented as a wrapper. Defining `Select` as we have here allows us to halve the number of times we have to deserialize the values when comparing multiple values.

Wrapper:

```go
func Less(key string, values [][]byte, v Validator) func(i, j int) bool { /* ... */ }
```